src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -121,9 +169,11 @@ const useFetchChannelHook = () => {
   const saveAllChannelData = async (channels, channelCategory: ChannelCategory) => {
     const filteredChannels = filterChannels(channels);
 
-    filteredChannels?.forEach(async (channel) => {
-      await saveChannelData(channel, channelCategory);
+    const channelPromises = filteredChannels?.map((channel) => {
+      return Promise.resolve(saveChannelData(channel, channelCategory));
     });
+
+    await Promise.all(channelPromises);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made to this section are good for performance. Instead of using `forEach` with an async function which could potentially block the execution, the new code uses `map` to create an array of promises and then waits for all of them to resolve using `Promise.all`. This change should improve the overall performance of the application.



<!-- This is an auto-generated comment by OSS CodeRabbit -->